pkg/virt-api/rest/subresource.go

```diff
@@ -908,48 +908,26 @@ func (app *SubresourceAPIApp) FilesystemList(request *restful.Request, response
 }
 
 func generateVMVolumeRequestPatch(vm *v1.VirtualMachine, volumeRequest *v1.VirtualMachineVolumeRequest) (string, error) {
-	verb := "add"
-	if len(vm.Status.VolumeRequests) > 0 {
-		verb = "replace"
-	}
-
+	verb := getPatchVerb(vm.Status.VolumeRequests)
 	vmCopy := vm.DeepCopy()
 
 	// We only validate the list against other items in the list at this point.
 	// The VM validation webhook will validate the list against the VMI spec
 	// during the Patch command
 	if volumeRequest.AddVolumeOptions != nil {
-		name := volumeRequest.AddVolumeOptions.Name
-		for _, request := range vm.Status.VolumeRequests {
-			if request.AddVolumeOptions != nil && request.AddVolumeOptions.Name == name {
-				return "", fmt.Errorf("Add volume request for volume [%s] already exists", name)
-			} else if request.RemoveVolumeOptions != nil && request.RemoveVolumeOptions.Name == name {
-				return "", fmt.Errorf("Unable to add volume. A remove volume request for volume [%s] already exists and is still being processed.", name)
-			}
+		if err := addAddVolumeRequests(vm, volumeRequest, vmCopy); err != nil {
+			return "", err
```

Comment:
I feel like `addAddVolumeRequests` is still doing more than 1 thing: validation and adding the request.
I think it may be better to do it like:

```
for _, request := range vm.Status.VolumeRequests {
    validateAddVolumeRequest(request)
}
vmCopy.Status.VolumeRequests = append(vm.Status.VolumeRequests, *volumeRequest)
```

Reply:
IIUC that's what I did in the last update, I just named the function `handleAddVolumeRequest` instead of `validateAddVolumeRequest`. Please have a look at line 962, it should be the same thing you are suggesting.

Reply:
Yes you're right, could we rename this function to `validateAddVolumeRequest` please? It sounds closer to what it's doing to me

Reply:
Sure, that sounds better